### PR TITLE
fix(init): keep user files in legacy command folders

### LINKED ISSUE #1873: `openspec init` deletes user files inside a legacy `commands/openspec/` folder, even without `--force`
## Description

Legacy cleanup treats six tools' old command folders as entirely OpenSpec's:
`.claude/commands/openspec/`, `.codebuddy/…`, `.qoder/…`, `.lingma/…`, `.crush/…`
and `.gemini/commands/openspec/`. If the folder exists, it is removed
recursively, whatever else is inside it.

People keep their own slash commands in that folder: it is the natural place for
a project's commands under an `openspec` namespace. Every file there is deleted
along with OpenSpec's three old commands. The upgrade prompt lists the folder
under "Files to remove / No user content to preserve", and the summary says only
`✓ Removed .claude/commands/openspec/`, so the user is never told their files
were in it.

`openspec init` runs this cleanup without asking when `--force` is set **or when
there is no TTY**, so an agent or CI running plain `openspec init --tools claude`
deletes the files with no prompt at all. `openspec update --force` goes through
the same function.

## Reproduction

```bash
mkdir -p .claude/commands/openspec
for f in proposal apply archive; do
  printf -- '---\nname: OpenSpec: %s\n---\nold\n' "$f" > .claude/commands/openspec/$f.md
done
printf -- '---\ndescription: my team review checklist\n---\nReview carefully.\n' \
  > .claude/commands/openspec/team-review.md

openspec init --tools claude < /dev/null     # no TTY, no --force
ls .claude/commands/openspec
```

## Expected

Only the files OpenSpec wrote (`proposal.md`, `apply.md`, `archive.md`) are
removed. `team-review.md` survives, and the summary says it was kept. The folder
goes only if nothing else is left in it.

## Actual

```
  init>   • .claude/commands/openspec/
  init> ✔ Legacy files cleaned up
  init>   ✓ Removed .claude/commands/openspec/ (replaced by OpenSpec skills and commands)
ls: cannot access '.claude/commands/openspec': No such file or directory
```

`team-review.md` is gone. The same happens with `--force`. With only OpenSpec's
three files in the folder, removing it is correct.

## Root cause

- `src/core/legacy-cleanup.ts:36-41` declares the six folders as `type: 'directory'`.
- Detection flags a folder whenever it exists (`:321-323`).
- Cleanup runs `fs.rm(fullPath, { recursive: true, force: true })` (`:555`).
- `src/core/init.ts:500-501` runs the cleanup automatically when
  `this.force || !canPrompt`.

The file already documents this hazard at `:69-71` for CoStrict, which was made
file-scoped in #1492. The directory entries never got the same treatment.

## Impact

Irreversible loss of user-authored files, triggered by the command users run to
upgrade, and by agents and CI without any prompt. `docs/cli.md` describes
`--force` as "Auto-cleanup legacy **files** without prompting", and the migration
guide says "Your own content is never deleted".

## Environment

- OpenSpec 1.13.0, reproduced on the published npm package and on a build of
  `main` @ `9d4e597`
- Node 20.19.0 and 22, Linux

### PR DESCRIPTION
**Pull request by dwin-gharibi**
Closes #1873.

## Why

Six entries in `LEGACY_SLASH_COMMAND_PATHS` (`src/core/legacy-cleanup.ts:36-41`)
were `directory` entries: Claude Code, CodeBuddy, Qoder, Lingma, Crush and
Gemini CLI, each at `<tool>/commands/openspec/`. Detection flagged the folder
whenever it existed (`:321-323`). Cleanup then ran
`fs.rm(fullPath, { recursive: true, force: true })` (`:555`), deleting everything
in it. Users keep their own commands in that folder, and they went with
OpenSpec's three old ones:

- The upgrade prompt listed the folder under "Files to remove / No user content
  to preserve".
- The summary said `✓ Removed .claude/commands/openspec/`.

So nothing ever told the user their files had been in it.

`openspec init` runs this cleanup automatically when `--force` is set **or when
there is no TTY** (`src/core/init.ts:500-501`). So an agent or CI running plain
`openspec init --tools claude` deleted the files without a prompt.
`openspec update --force` uses the same function.

The file already names the hazard for CoStrict (`:69-71`), which #1492 made
file-scoped. The directory entries never got the same treatment.

## What Changes

- **Each directory entry now lists the files OpenSpec wrote there**, in a new
  `managedFileNames`. The names come from the slash configurators removed in
  #565: `proposal.md`, `apply.md` and `archive.md`, or `.toml` for Gemini.
  Lingma is the exception. Its support arrived after the opsx rename and has
  always written to `.lingma/commands/opsx/`, so OpenSpec never wrote a file
  into `.lingma/commands/openspec/`, and its list is empty.
- **Detection:**
  - A folder holding only those files, or nothing, is reported as a folder,
    exactly as before.
  - A folder that also holds anything else has only OpenSpec's files reported,
    one by one. The upgrade prompt then lists exactly what will be deleted.
  - A folder holding none of OpenSpec's files is not reported at all.
- **Cleanup deletes only managed files.** It then removes the folder with a
  non-recursive `rmdir`, and only if the folder is empty. Anything left is
  recorded in a new optional `CleanupResult.keptFiles` and printed as
  `• Kept .claude/commands/openspec/team-review.md (not created by OpenSpec)`.
  This also protects a file added between detection and cleanup, for example
  while the interactive prompt waits.
- **Files reported from a mixed folder map back to their tool.** So
  `getToolsFromLegacyArtifacts` and `omitToolLegacyArtifacts` treat them like any
  other legacy file. The second is what the legacy-upgrade path uses to skip a
  tool whose replacement was not written.
- **Ownership is checked by content, not just by name.** A managed file counts as OpenSpec's only when it is a regular file whose content still carries the OpenSpec markers every legacy command was generated with. Cleanup re-checks that right before each unlink, so a same-named file the user wrote (or swapped in while the prompt waited) is kept. A symlinked command folder is never followed.

The common upgrade is unchanged: a folder holding only OpenSpec's files is
removed as before, with the same prompt line and the same summary line.

## Testing

New file `test/core/legacy-cleanup.user-files.test.ts`, now 30 tests. The first
24 were written first: on clean `9d4e597` they gave 15 failed and 9 passed, every
failure a bug assertion and every control passing. The later ownership and
recheck tests each fail without the change that added them. With the fix, all 30
pass.

Edge cases covered:

- **Per tool (claude, codebuddy, qoder, crush, gemini):** a folder of only
  OpenSpec files is still removed (control). A user file in it is kept, the
  OpenSpec files are deleted, the folder stays, and the files map back to the
  tool.
- A nested folder of user commands is kept, and so is a folder named like a
  legacy file (`apply.md/`).
- A Gemini `proposal.md` is kept while `proposal.toml` is removed.
- A folder holding no OpenSpec files is neither reported nor touched.
- Lingma files are left alone. An empty leftover folder is still removed.
- A file added between detection and cleanup is kept.
- A user-authored file that only shares a legacy command name is kept, alone or
  beside OpenSpec's files, and a symlinked command folder is never followed.
- A `proposal.md` the user swaps in between detection and cleanup, or after
  cleanup has scanned the folder, is kept and reported as kept.
- The upgrade prompt lists OpenSpec's files, not the folder. The summary names
  what was kept and never claims the folder was removed.
- When `omitToolLegacyArtifacts` skips the tool, a mixed folder is untouched.
- A guard fails if a new directory entry is added without a test row.
- **End to end:** `openspec init --tools claude` keeps `team-review.md`, both
  without a TTY and with `--force`. It still removes a folder of only OpenSpec
  files.

Two existing tests encoded the old behaviour and were updated:

- **`test/core/update.test.ts`** › "should cleanup legacy slash command
  directories with --force". Its fixture was `old-command.md`, a file OpenSpec
  never wrote, and it asserted that such a file is deleted along with the folder.
  That is the behaviour this PR removes. The fixture is now `proposal.md`, and
  the assertions are unchanged.
- **`test/core/legacy-cleanup.test.ts`** › "should include expected tool
  patterns". It compares the claude entry with `toEqual`, so it now includes
  `managedFileNames`.

## Verification

Run in a Linux sandbox under Node 20.19.0, the CI version, on `9d4e597` with
this patch:

- `pnpm run build`: ok
- `pnpm exec tsc --noEmit`: ok
- `pnpm lint`: ok
- Targeted (`legacy-cleanup.user-files`, `legacy-cleanup`, `update`, `init`):
  all passed
- Full suite, `VITEST_MAX_WORKERS=4 pnpm test`: 4579 passed and 8 failed, in 5
  files. The 5 files are `store-references`, `store-root-selection`, `store`,
  `workset` and `package-install-scripts`, none of which touches legacy
  cleanup. Every failure was a 10-second test timeout on a shared, heavily
  loaded machine (load average 6 to 13).

None of these failures come from this change:

- `store-references`, `store-root-selection`, `workset` and
  `package-install-scripts` fail the same tests, the same way, on clean
  `9d4e597` under the same load.
- `store` passes 43/43 on this branch and on clean `9d4e597` when run
  side by side. Its two slow tests take 8.6 to 9.0 seconds even on
  `9d4e597`.
- Only `init.ts` and `update.ts` import `legacy-cleanup.ts`, and none of the
  store or workset commands reach either one.

## Changeset

Added `.changeset/legacy-cleanup-keeps-user-files.md` (patch).

## Summary by CodeRabbit

* **Bug Fixes**
  * Legacy cleanup now removes only files generated by OpenSpec, preserving user-created files with matching names.
  * Symlinked command folders and files without OpenSpec markers are protected from removal.
  * Cleanup summaries now list retained files and remove legacy folders only when empty.

### CODE REVIEW

File: src/core/legacy-cleanup.ts [hunk 1]
```diff
@@ -335,6 +357,76 @@ export async function detectLegacySlashCommands(
...
+  try {
+    entries = await fs.readdir(dirPath, { withFileTypes: true });
+  } catch {
+    return undefined;
+  }
+
+  const managed: string[] = [];
+  const others: string[] = [];
+  for (const entry of entries) {
+    if (entry.isFile() && managedFileNames.includes(entry.name)) {
```
[the remark this answers is not in the document]
Comment (clay-good) [reply src/core/legacy-cleanup.ts]: Valid, fixed in 5c5462e2. A legacy command file now counts as OpenSpec's only if it is a regular file with a managed name whose content still carries the OpenSpec markers (every legacy slash command was generated with them, and OpenSpec refused to update one without them). Cleanup re-checks that immediately before each unlink, so a proposal.md replaced between detection and cleanup is kept and reported. Symlinked command folders are never followed. Regression tests cover both replacement cases, a same-named user file, and the symlink case; all fail on the previous head.

**Comment by clay-good** [comment]
Hardening pushed for merge:

- 5c5462e2 fix(init): a legacy command file now counts as OpenSpec's only when it is a regular file with a managed name whose content still carries the OpenSpec markers (every legacy command was generated with them). Cleanup re-checks this right before each unlink, and a symlinked command folder is never followed. This resolves the CodeRabbit ownership thread. Existing update/legacy-cleanup fixtures now use marker-wrapped content like real generated files. The new tests (same-named user file, replacement between detection and cleanup, symlinked folder) fail on the previous head.
- 63cf6525 docs: dropped the docs/migration-guide.md edit. docs/ is legacy and the canonical docs-lab page (help/legacy/migration.md) is still a skeleton.

Verified: build, tsc, lint; legacy-cleanup, legacy-cleanup.user-files, update, init tests 417/417. A real `openspec init --tools claude` repro on main deleted the whole folder; now only the generated file goes.

**Comment by clay-good** [comment]
Second pass, answering CodeRabbit's outside-diff finding at `src/core/legacy-cleanup.ts:684` (revalidate ownership before each directory-file deletion): confirmed. The per-file loop rechecked before each unlink, but the directory loop unlinked every name the scan had classified without checking again, so a file swapped in after that scan was deleted and reported as deleted.

Fixed in cf9e9cc1: each file in the directory loop is now checked for the OpenSpec markers immediately before its unlink. Only files that pass are recorded as deleted; a file that fails is left in place, so the folder is not removed and the file is reported under "Kept".

Regression test: `keeps proposal.md when the user replaces it after cleanup has scanned the folder` in `test/core/legacy-cleanup.user-files.test.ts`. It swaps in a user's `proposal.md` right after cleanup's own scan reads the generated one. It fails without the fix and passes with it.

Also re-verified the marker assumption for every directory entry against the old generators: Claude Code, CodeBuddy, Qoder and Crush wrote markdown through the shared base configurator, and Gemini CLI wrote `.toml` with the markers inside the `prompt` string, in every version from its first commit. All of them refused to update a file that had lost its markers. Lingma never wrote into `commands/openspec/`.

**Review by alfred-openspec — APPROVED** [review]
Legacy cleanup now identifies OpenSpec-owned files by both name and markers, rechecks ownership before unlinking, never follows linked command folders, and removes directories only when empty. Focused cleanup and update tests pass; the current canonical docs remain accurate because they do not promise recursive legacy-folder deletion.
